keps/sig-node/4033-group-driver-detection-over-cri/README.md

```diff
@@ -0,0 +1,892 @@
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+If either kubelet or the container runtime running on the node does not support
+the new field in the CRI API, they just resort to the existing behavior of
+respecting their individual cgroup-driver setting. That is, if the node has a
+container runtime that does not support this field the kubelet will use its
+cgroupDriver setting from kubeletConfig (or `--cgroup-driver` commandline
+flag).  This is also the case if the kubelet does not support the new field:
+the information about cgroup driver advertised by the runtime will be just
+ignored by kubelet and it will resort to its own configuration settings. Note:
+this does present a configuration skew risk, but that risk is the same as
+currently exists today.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+No feature gate required–the fields are all SIG Node internal and have simple
+fallbacks.
```

Comment:
Also point out that this feature is only enabled when used with a CRI using the new field? So there is a reasonable level of enablement control.

Reply:
@jpbetz @pacoxu  with the addition of a feature gate this was now changed to 

```
- [X] Feature gate (also fill in values in `kep.yaml`)
  - Feature gate name: KubeletCgroupDriverFromCRI
  - Components depending on the feature gate: kubelet
```